packages/components/text/src/text.tsx

```diff
@@ -0,0 +1,284 @@
+import type { MessageDescriptor } from 'react-intl';
+
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useTheme } from '@emotion/react';
+import invariant from 'tiny-invariant';
+import { oneLine } from 'common-tags';
+import {
+  filterDataAttributes,
+  warnDeprecatedProp,
+} from '@commercetools-uikit/utils';
+import {
+  bodyStyles,
+  detailStyles,
+  headlineStyles,
+  subheadlineStyles,
+  wrapStyles,
+} from './text.styles';
+
+type TBasicTextProps = {
+  intlMessage?: MessageDescriptor;
+  children?: React.ReactNode;
+};
+
+type TBasicHeadlineProps = {
+  title?: string;
+};
+
+const getIsIntlMessage = (
+  intlMessage: unknown
+): intlMessage is MessageDescriptor => {
+  if (intlMessage && typeof intlMessage === 'object') {
+    return (
+      'id' in intlMessage &&
+      'description' in intlMessage &&
+      'defaultMessage' in intlMessage
+    );
+  }
+  return false;
+};
+
+const warnIfMissingTitle = (
+  props: TBasicHeadlineProps,
+  componentName: string
+) => {
+  if (typeof props.title === 'string') {
+    invariant(
+      props.title.length > 0,
+      `Invalid prop 'title' supplied to '${componentName}'. Expected it to be nonempty string, but it was empty.`
+    );
+  }
+};
+
+const warnIfMissingContent = (
+  props: TBasicTextProps,
+  componentName: string
+) => {
+  const hasContent =
+    getIsIntlMessage(props.intlMessage) ||
+    Boolean(React.Children.count(props.children));
+
+  invariant(
+    hasContent,
+    oneLine`
```

Comment:
Maybe we can just drop this dependency (and thus the related dep type)? It feels like a bit unnecessary...

Reply:
My preference is to deconstruct a long string into multi lines so that it is easier to read, and I value that.
I propose we keep it

Reply:
You can simply do

```js
[
  'This is the first line',
  'Then a second',
  'and a third.'
].join('\n');
```

I just want to avoid having unnecessary dependencies.